src/positron-dts/positron.d.ts

```diff
@@ -648,13 +648,27 @@ declare module 'positron' {
 
 	export interface LanguageRuntimeManager {
 		/**
-		 * Returns a generator that yields metadata about the language runtimes
-		 * that are available to the user.
+		 * Returns a generator that yields metadata about all the language
+		 * runtimes that are available to the user.
 		 *
 		 * This metadata will be passed to `createSession` to create new runtime
 		 * sessions.
 		 */
-		discoverRuntimes(): AsyncGenerator<LanguageRuntimeMetadata>;
+		discoverAllRuntimes(): AsyncGenerator<LanguageRuntimeMetadata>;
+
+		/**
+		 * Returns a single runtime metadata object representing the runtime
+		 * that should be used in the current workspace, if any.
+		 *
+		 * Note that this is called before `discoverAllRuntimes` during
+		 * startup, and should return `undefined` if no runtime is recommended.
+		 *
+		 * If a runtime is returned, `startupBehavior` property of the runtime
+		 * metadata is respected here; use `Immediately` to start the runtime
+		 * right away, or any other value to save the runtime as the project
+		 * default without starting it.
+		 */
+		recommendedWorkspaceRuntime(): Thenable<LanguageRuntimeMetadata | undefined>;
```

Comment:
Checking my understanding! Does this sound right?

Extensions will determine this recommended runtime, likely with this order of precedence:

1. Has the user indicated a default runtime at the Workspace level? i.e. `python.defaultInterpreterPath`, an upcoming `r.interpreters.default` option; startupBehavior `Immediately`
2. Is anything else in this workspace indicating a preferred runtime? i.e. Venv, DESCRIPTION, etc.; startupBehavior `Immediately`
3. Consult User-level setting for default runtime? i.e. `python.defaultInterpreterPath`, an upcoming `r.interpreters.default` option; startupBehavior `Implicit`